src/Core/Services/Implementations/StripeSyncService.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace Bit.Core.Services
+{
+    public class StripeSyncService : IStripeSyncService
+    {
+        private readonly IStripeAdapter _stripeAdapter;
+
+        public StripeSyncService(IStripeAdapter stripeAdapter)
+        {
+            _stripeAdapter = stripeAdapter;
+        }
+
+        public async Task UpdateCustomerEmailAddress(string gatewayCustomerId, string emailAddress)
+        {
+            var customer = await _stripeAdapter.CustomerGetAsync(gatewayCustomerId);
+
+            await _stripeAdapter.CustomerUpdateAsync(customer.Id,
+                new Stripe.CustomerUpdateOptions { Email = emailAddress });
+        }
```

Comment:
Only thing I'd add is checking for `IsNullOrWhiteSpace` on both parameters (invalid gatewayCustomerId and would not want a customer to update their email to `nothing`. This is to ensure we don't cause a NullRef, also what happens when no customer is found/returned. `customer.Id` would throw.

In all mentioned cases we'd want to throw, so we can revert the change, but also give a specific reason for the failure and not just a generic NullRefEx.

Reply:
Yeah I should have checked for those